pyro/infer/mcmc/util.py

```diff
@@ -303,32 +303,32 @@ def get_potential_fn(self, jit_compile=False, skip_jit_warnings=True, jit_option
 def _get_init_params(model, model_args, model_kwargs, transforms, potential_fn, prototype_params,
                      max_tries_initial_params=100, num_chains=1, strategy="uniform"):
     params = prototype_params
-    params_per_chain = defaultdict(list)
-    n = 0
 
     # For empty models, exit early
     if not params:
         return params
 
-    for i in range(max_tries_initial_params):
-        while n < num_chains:
-            if strategy == "uniform":
-                params = {k: dist.Uniform(v.new_full(v.shape, -2), v.new_full(v.shape, 2)).sample()
-                          for k, v in params.items()}
-            elif strategy == "prior":
-                trace = poutine.trace(model).get_trace(*model_args, **model_kwargs)
-                samples = {name: trace.nodes[name]["value"].detach() for name in params}
-                params = {k: transforms[k](v) for k, v in samples.items()}
-            pe_grad, pe = potential_grad(potential_fn, params)
-
-            if torch.isfinite(pe) and all(map(torch.all, map(torch.isfinite, pe_grad.values()))):
-                for k, v in params.items():
-                    params_per_chain[k].append(v)
-                n += 1
-        if num_chains == 1:
-            return {k: v[0] for k, v in params_per_chain.items()}
-        else:
-            return {k: torch.stack(v) for k, v in params_per_chain.items()}
+    params_per_chain = defaultdict(list)
+    num_found = 0
+    for attempt in range(num_chains * max_tries_initial_params):
```

Comment:
I think the difference is at this line: before, we use `max_tries_initial_params`, now we use `num_chains * max_tries_initial_params`?

Reply:
> The original logic tries to gather a set of size num_chains each attempt

No, we didn't reset `n` after exiting the `while` loop.

Reply:
Opss, I get your point. The old logic is wrong (my bad).

Reply:
Oh, I see, thanks for pointing that out!